plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/BaseClickHouseTypeMapping.java

```diff
@@ -688,16 +692,31 @@ public void testTimestamp(ZoneId sessionZone)
                 .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(sessionZone.getId()))
                 .build();
 
-        SqlDataTypeTest.create()
-                .addRoundTrip("timestamp(0)", "timestamp '1986-01-01 00:13:07'", createTimestampType(0), "TIMESTAMP '1986-01-01 00:13:07'") // time gap in Kathmandu
-                .addRoundTrip("timestamp(0)", "timestamp '2018-03-25 03:17:17'", createTimestampType(0), "TIMESTAMP '2018-03-25 03:17:17'") // time gap in Vilnius
-                .addRoundTrip("timestamp(0)", "timestamp '2018-10-28 01:33:17'", createTimestampType(0), "TIMESTAMP '2018-10-28 01:33:17'") // time doubled in JVM zone
-                .addRoundTrip("timestamp(0)", "timestamp '2018-10-28 03:33:33'", createTimestampType(0), "TIMESTAMP '2018-10-28 03:33:33'") // time double in Vilnius
-                .execute(getQueryRunner(), session, trinoCreateAsSelect(session, "test_timestamp"))
+        // ClickHouse DateTime64 has sub-second precision
+        SqlDataTypeTest timestampTests = SqlDataTypeTest.create();
+        IntStream.rangeClosed(0, 9).forEach(precision -> {
```

Comment:
replace 9 here and elsewhere with `CLICKHOUSE_MAX_SUPPORTED_DATETIME64_PRECISION`?

Reply:
@tangjiangling would you be able to act on this one or else I can help with that. Thanks